src/ctia/http/routes/crud.clj

```diff
@@ -229,45 +231,59 @@
                 :auth-identity identity
                 :identity-map identity-map
                 (GET "/histogram" []
-                     :return [{:key s/Str :value s/Int}]
+                     :return AggResult
                      :summary (format "Histogram for a %s field" capitalized)
                      :query [params histogram-q-params]
-                     (ok (read-store
-                          entity
-                          aggregate
-                          (search-query (keyword
-                                         (:aggregate-on params))
-                                        (st/select-schema params agg-search-schema)
-                                        coerce-from)
-                          (st/assoc (st/select-schema params HistogramParams)
-                                    :agg-type :histogram)
-                          identity-map)))
+                     (let [aggregate-on (keyword (:aggregate-on params))
```

Comment:
I couldn't see the `histogram-q-params` but, my guess is that it might be quite open.
I would really take care about using `keyword` here.
Because `keyword` is quite fragile on strings containing special chars.
So if user can provide any string here, I would take extra precaution to prevent any form of attack.

for example `(keyword "abc def") => :abc def` or `(keyword "123abc") => :123acb` etc...

Reply:
`histogram-q-params` is [built-on](https://github.com/threatgrid/ctia/pull/896/files#diff-e934e438d94b4fd1a73d3a138adced8dR92) `search-q-params`, it keeps entity filters and adds histogram specific params.
`aggregate-on` possible values is defined per entity type, and should not contain weird values, since its fields or fields + subfields.
Here is the possible values for [incident](https://github.com/threatgrid/ctia/pull/896/files#diff-af4c6ba2f07524c30fea358b19b8beeeR177):
``` clojure
(def incident-histogram-fields
  [:timestamp
   :incident_time.opened
   :incident_time.discovered
   :incident_time.reported
   :incident_time.remediated
   :incident_time.closed
   :incident_time.rejected])
```